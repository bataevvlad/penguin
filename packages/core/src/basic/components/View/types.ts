import {ViewProps as DefaultViewProps, View as DefaultView} from 'react-native';

export type IViewProps = DefaultViewProps;
export type IView = DefaultView;
