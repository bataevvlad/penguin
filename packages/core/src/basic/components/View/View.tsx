import {forwardRef} from 'react';
import {View as DefaultView} from 'react-native';

import {IView, IViewProps} from './types';

// eslint-disable-next-line react/display-name
const View = forwardRef<IView, IViewProps>((props, ref) => (
  <DefaultView ref={ref} {...props} />
));

export default View;
