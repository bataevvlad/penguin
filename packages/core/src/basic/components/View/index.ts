import View from './View';

export * from './types';

export {View};
