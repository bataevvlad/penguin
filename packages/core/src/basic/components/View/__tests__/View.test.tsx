import {render} from '@testing-library/react-native';

import View from '../View';

describe('@penguin/core/View', () => {
  it('renders correctly', () => {
    const {toJSON} = render(<View />);
    expect(toJSON()).toMatchSnapshot();
  });
});
