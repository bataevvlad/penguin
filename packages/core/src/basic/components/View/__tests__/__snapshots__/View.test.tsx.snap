// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@penguin/core/View renders correctly 1`] = `<View />`;
